tapir/coop/models.py

```diff
@@ -719,3 +718,63 @@ def populate(
             old_frozen=old_frozen,
             new_frozen=new_frozen,
         )
+
+class ResignedMembership(models.Model):
+    share_owner = models.ForeignKey(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name=_("Shareowner"), related_name="share_owner",
+    )
+    cancellation_date = models.DateField(
+        default=timezone.now,
+        blank=True,
+        )
+    pay_out_day = models.DateField(null=True)
+    cancellation_reason = models.CharField(max_length = 1000)
+    coop_buys_shares_back = models.BooleanField()
+    willing_to_gift_shares_to_coop = models.BooleanField()
+    transfering_shares_to = models.ForeignKey(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name="OwnerToTransfer", null=True, related_name="owner_to_transfer",
+    )
+    paid_out = models.BooleanField(default=False)
+
+    class ResignedMemberQuerySet(models.QuerySet):
+        def with_term(self, search_string: str):
+            searches = [s for s in search_string.split(" ") if s != ""]
+
+            for search in searches:
+                word_filter = (
+                    Q(share_owner__user__usage_name__unaccent__icontains=search)
+                    | Q(share_owner__user__first_name__unaccent__icontains=search)
+                    | Q(share_owner__user__last_name__unaccent__icontains=search)
+                    | Q(share_owner__id__icontains=search)
+                )
```

Comment:
I see 2 problems here. They would become irrelevant we use `ShareOwner.objects.with_name`, but so you know:

- You only search within `share_owner__user`, but if a member never got a TapirUser object, like is the case for most investing members for example, you won't find them. That's why `ShareOwnerQuerySet.with_name` searches for both `last_name__unaccent__icontains` and `user__last_name__unaccent__icontains`
- You overwrite the `word_filter` variable on every step of the for loop. So if there are several elements in `searches`, only the last one will be used. I think you intend to combine instead of overwrite.